### PR TITLE
trace-agent: get rid of the statsd singleton

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

AIT-9770

- Remove the statsd singleton in the trace agent. Inject the statsd client as a dependency and propagate it to all the trace-agent components instead.
- Remove `pkg/trace/metrics`

### Motivation

- [Singletons are bad for many reasons](https://stackoverflow.com/questions/137975/what-are-drawbacks-or-disadvantages-of-singleton-pattern)
- Better support for trace-agent as a library

### Describe how to test/QA your changes

The new-e2e tests make assertions on the statsd metrics emitted by the trace-agent :)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/serverless/trace/trace.go [hunk 1]
```diff
@@ -95,7 +97,7 @@ func (s *ServerlessTraceAgent) Start(enabled bool, loadConfig Load, lambdaSpanCh
 			context, cancel := context.WithCancel(context.Background())
 			tc.Hostname = ""
 			tc.SynchronousFlushing = true
-			s.ta = agent.NewAgent(context, tc, telemetry.NewNoopCollector())
+			s.ta = agent.NewAgent(context, tc, telemetry.NewNoopCollector(), &statsd.NoOpClient{})
```
Comment (ahmed-mez) [inline pkg/serverless/trace/trace.go:100]: This keeps the behavior prior to this PR. The statsd client in `metrics` was no-op in serverless.

**Review by knusbaum — APPROVED** [review]
Thanks, looks good.

**Review by liustanley — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

**Review by DylanLovesCoffee — APPROVED** [review]

**Pull request merged**